server/test/unit/server/async/test_app.py

```diff
@@ -16,17 +24,87 @@ class InitializeWorkerTestCase(unittest.TestCase):
+    @mock.patch('pulp.server.async.app.get_resource_manager_lock')
+    def test_initialize_worker(self,
+                               mock_get_resource_manager_lock,
+                               _delete_worker, initialize,
+                               create_worker_working_directory,
                                delete_worker_working_directory):
         """
         Assert that initialize_worker() calls Pulp's initialization code and the appropriate worker
-        monitoring code.
+        monitoring code for a non-resource mananger worker.
+        """
+        sender = 'reserved_resource_worker-0' + '@' + platform.node()
+        # The instance argument isn't used and don't matter, so we'll just pass a mock
+        app.initialize_worker(sender, mock.MagicMock())
+
+        initialize.assert_called_once_with()
+        _delete_worker.assert_called_once_with(sender, normal_shutdown=True)
+        create_worker_working_directory.assert_called_once_with(sender)
+        delete_worker_working_directory.assert_called_once_with(sender)
+        mock_get_resource_manager_lock.assert_not_called()
+
+    @mock.patch('pulp.server.async.app.common_utils.delete_worker_working_directory')
+    @mock.patch('pulp.server.async.app.common_utils.create_worker_working_directory')
+    @mock.patch('pulp.server.async.app.initialization.initialize')
+    @mock.patch('pulp.server.async.app.tasks._delete_worker')
+    @mock.patch('pulp.server.async.app.get_resource_manager_lock')
+    def test_initialize_worker_resource_manager(self,
+                                                mock_get_resource_manager_lock,
+                                                _delete_worker, initialize,
+                                                create_worker_working_directory,
+                                                delete_worker_working_directory):
+        """
+        Assert that initialize_worker() calls Pulp's initialization code and the appropriate worker
+        monitoring code for a resource mananger worker.
         """
-        sender = mock.MagicMock()
-        # The args aren't used and don't matter, so we'll just pass some mocks
+        sender = RESOURCE_MANAGER_WORKER_NAME + '@' + platform.node()
+        # The instance argument isn't used and don't matter, so we'll just pass a mock
         app.initialize_worker(sender, mock.MagicMock())
 
         initialize.assert_called_once_with()
         _delete_worker.assert_called_once_with(sender, normal_shutdown=True)
         create_worker_working_directory.assert_called_once_with(sender)
         delete_worker_working_directory.assert_called_once_with(sender)
+        mock_get_resource_manager_lock.assert_called_once_with(sender)
+
+    @mock.patch('pulp.server.async.app.datetime')
+    @mock.patch('pulp.server.async.app.Worker')
+    @mock.patch('pulp.server.async.app.ResourceManagerLock')
+    def test_get_resource_manager_lock(self, mock_rm_lock, mock_worker, mock_datetime):
```

Comment:
One last testing request: it would be particularly valuable to have a test that exercises the block of code under the ``except`` statement. Especially since the normal path for typical deployments will never hit that block, basic test coverage will be a good idea.

You can mock ``sleep()`` so it doesn't actually sleep, and make ``mock_rm_lock.objects().save.side_effects`` be a list so it raises an exception the first time, and not the second time.